client/src/config.rs

```diff
@@ -27,6 +27,10 @@ pub struct Configuration {
     pub private_key: PrivateKey,
     /// User account id.
     pub account_id: AccountId,
+    /// Login for basic auth. Should be used with `password`.
+    pub login: Option<SmallStr>,
+    /// Password for basic auth. Should be used with `login`.
+    pub password: Option<SmallStr>,
```

Comment:
logically, these could be grouped into a `Option<(SmallStr, SmallStr)>` and that way creating client couldn't result in an error. I grant that this might be inconvenient

Reply:
I like the suggestion to make it a `BasicAuth` struct. There's a subtlety that I hadn't initially considered: 
If you specify one field and not the other it's a malformed config file. 

But in the spec, both `password` and the `login` can be empty: base64: `Og==` for `null`. These would deserialise to the same as a malformed config.

Reply:
Which reminds me, neither field should have whitespace nor the `:` character in them. @Arjentix I would follow the way it's done in #1909 , for the `Name` type. Don't bother with providing unchecked variants, it's always better to `expect` and `panic`.

Reply:
Done! [RFC](https://datatracker.ietf.org/doc/html/rfc7617#section-2) says, that only `login` can't have `:` character.

I'm closing this conversation, cause it's outdated now